framework/decode/vulkan_replay_consumer_base.cpp

```diff
@@ -6283,6 +6358,50 @@ VkResult VulkanReplayConsumerBase::OverrideGetAndroidHardwareBufferPropertiesAND
     }
 }
 
+void VulkanReplayConsumerBase::ClearCommandBufferInfo(CommandBufferInfo* command_buffer_info)
+{
+    command_buffer_info->is_frame_boundary = false;
+}
+
+VkResult VulkanReplayConsumerBase::OverrideBeginCommandBuffer(
+    PFN_vkBeginCommandBuffer                                func,
+    VkResult                                                original_result,
+    CommandBufferInfo*                                      command_buffer_info,
+    StructPointerDecoder<Decoded_VkCommandBufferBeginInfo>* begin_info_decoder)
+{
+    ClearCommandBufferInfo(command_buffer_info);
+
+    VkCommandBuffer                 command_buffer = command_buffer_info->handle;
+    const VkCommandBufferBeginInfo* begin_info     = begin_info_decoder->GetPointer();
+    return func(command_buffer, begin_info);
+}
+
+VkResult VulkanReplayConsumerBase::OverrideResetCommandBuffer(PFN_vkResetCommandBuffer  func,
+                                                              VkResult                  original_result,
+                                                              CommandBufferInfo*        command_buffer_info,
+                                                              VkCommandBufferResetFlags flags)
+{
+    ClearCommandBufferInfo(command_buffer_info);
+
+    VkCommandBuffer command_buffer = command_buffer_info->handle;
+    return func(command_buffer, flags);
+}
+
+void VulkanReplayConsumerBase::OverrideCmdDebugMarkerInsertEXT(
+    PFN_vkCmdDebugMarkerInsertEXT                             func,
+    CommandBufferInfo*                                        command_buffer_info,
+    StructPointerDecoder<Decoded_VkDebugMarkerMarkerInfoEXT>* marker_info_decoder)
+{
+    const VkDebugMarkerMarkerInfoEXT* marker_info = marker_info_decoder->GetPointer();
+    func(command_buffer_info->handle, marker_info);
+
+    // Look for the debug marker that identifies this command buffer as a VR frame boundary.
+    if (util::platform::StringContains(marker_info->pMarkerName, "vr-marker,frame_end,type,application"))
```

Comment:
This string literal is used in encode too. Should we define it once in a file seen by both? `"vr-marker,frame_end,type,application")`

Also, does the match need to account for case and whitespace?